src/stripe/stripe.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Inject, Injectable } from '@nestjs/common';
+import Stripe from 'stripe';
+
+@Injectable()
+export class StripeService {
+  private stripe: Stripe;
+
+  constructor(@Inject('STRIPE_API_KEY') private readonly apiKey: string) {
+    this.stripe = new Stripe(this.apiKey, {
+      apiVersion: '2024-04-10',
+    });
```

Comment:
same as earlier, we should have this service built by a factory so that you can only inject the built service as the dependency